app/controllers/miq_ae_class_controller.rb

```diff
@@ -172,6 +172,9 @@ def get_node_info(node)
       get_method_node_info(id)
     when "aen"
       @record = MiqAeNamespace.find_by_id(from_cid(id[1]))
+      # need to set record as Domain record if it's a domain, editable_domains, enabled_domains,
+      # visible domains methods returns list of Domains, need this for toolbars to hide/disable correct records.
+      @record = MiqAeDomain.find_by_id(from_cid(id[1])) if @record.domain?
```

Comment:
Since both kind of records live in miq_ae_namespace table, i needed to convert the class of record to be MiqAeDomain if record is domain, not sure how else i could switch the class of record from MiqAeNamespace to MiqAeDomain, when a link from tree in UI is pressed it passes in same prefix for the namespace/domain nodes, for example tree node id for domain or namespace will both have a prefix 'aen'